test/JoinTest.cpp

```diff
@@ -230,11 +238,27 @@ using ExpectedColumns = ad_utility::HashMap<
     std::pair<std::span<const Id>, ColumnIndexAndTypeInfo::UndefStatus>>;
 
 // Test that the result of the `join` matches the `expected` outcome.
-void testJoinOperation(Join& join, const ExpectedColumns& expected) {
-  auto res = join.getResult();
+void testJoinOperation(Join& join, const ExpectedColumns& expected,
+                       bool requestLaziness = false,
+                       bool expectLazinessParityWhenNonEmpty = false,
+                       ad_utility::source_location location =
+                           ad_utility::source_location::current()) {
+  auto lt = generateLocationTrace(location);
+  auto res = join.getResult(false, requestLaziness
+                                       ? ComputationMode::LAZY_IF_SUPPORTED
+                                       : ComputationMode::FULLY_MATERIALIZED);
   const auto& varToCols = join.getExternallyVisibleVariableColumns();
   EXPECT_EQ(varToCols.size(), expected.size());
-  const auto& table = res->idTable();
+  if (expectLazinessParityWhenNonEmpty &&
+      (!res->isFullyMaterialized() || !res->idTable().empty())) {
+    EXPECT_EQ(res->isFullyMaterialized(), !requestLaziness);
+  }
+  const auto& table = res->isFullyMaterialized()
+                          ? res->idTable()
+                          : static_cast<const IdTable&>(
+                                aggregateTables(std::move(res->idTables()),
+                                                join.getResultWidth())
+                                    .first);
```

Comment:
Does the ternary operator guarantee lifetime extensions for references to tempoararies in this case? Look up and comment:)